dora/core/common/src/main/java/alluxio/membership/WorkerServiceEntity.java

```diff
@@ -93,30 +95,31 @@ public int hashCode() {
     return Objects.hashCode(mAddress, mServiceEntityName);
   }
 
-  /**
-   * Serialize the WorkerServiceEntity object.
-   * @param dos
-   * @throws IOException
-   */
-  public void serialize(DataOutputStream dos) throws IOException {
-    super.serialize(dos);
-    dos.writeInt(mState.ordinal());
-    byte[] serializedArr = GrpcUtils.toProto(mAddress).toByteArray();
-    dos.writeInt(serializedArr.length);
-    dos.write(serializedArr);
+  @Override
+  public byte[] serialize() throws IOException {
+    try (ByteArrayOutputStream baos = new ByteArrayOutputStream();
+         DataOutputStream dos = new DataOutputStream(baos)) {
+      dos.writeUTF(mServiceEntityName);
+      dos.writeLong(mRevision);
+      dos.writeInt(mState.ordinal());
+      byte[] serializedArr = GrpcUtils.toProto(mAddress).toByteArray();
+      dos.writeInt(serializedArr.length);
+      dos.write(serializedArr);
+      return baos.toByteArray();
```

Comment:
actually i couldnt make a serializer working for both classes without introducing interfaces, and that adds too many layers just for the sake of serialization. when u rebase can make WorkerServiceEntity.serialize/deserialize to be first into jsonstr and then into byte[] ?

Reply:
np. you can merge #17959 first and i'll deal with the interface